From 5bfd2060b79719da1fb085f4b68a10fe91a86852 Mon Sep 17 00:00:00 2001
From: Valentin Iliushin <v.iliushin@partner.samsung.com>
Date: Mon, 22 May 2023 13:26:27 +0000
Subject: [PATCH 123/128] Adapt to SLS device changes

Adapt to changes in SlsDevice and PNMContext

Refers to: MCS23-974

Signed-off-by: Valentin Iliushin <v.iliushin@partner.samsung.com>
---
 aten/src/ATen/native/pnm/PNMDeviceContext.cpp | 4 ++--
 caffe2/operators/lengths_reducer_ops_pnm.cc   | 6 ++++--
 caffe2/python/pybind_state_pnm.cc             | 2 +-
 3 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/aten/src/ATen/native/pnm/PNMDeviceContext.cpp b/aten/src/ATen/native/pnm/PNMDeviceContext.cpp
index 60ceeb17..6c4784a3 100644
--- a/aten/src/ATen/native/pnm/PNMDeviceContext.cpp
+++ b/aten/src/ATen/native/pnm/PNMDeviceContext.cpp
@@ -7,12 +7,12 @@ namespace native {
 namespace pnm {
 
 Device& PnmDeviceContext::device() {
-  static auto static_device = SlsDevice::make();
+  static auto static_device = SlsDevice::make(PNMDevice::Type::SLS_AXDIMM);
   return static_device;
 }
 
 ::pnm::ContextHandler& PnmDeviceContext::context() {
-  static auto context = ::pnm::make_context(::pnm::ContextType::SLS);
+  static auto context = ::pnm::make_context(PNMDevice::Type::SLS_AXDIMM);
   return context;
 }
 
diff --git a/caffe2/operators/lengths_reducer_ops_pnm.cc b/caffe2/operators/lengths_reducer_ops_pnm.cc
index 3768212d..e235e0f0 100644
--- a/caffe2/operators/lengths_reducer_ops_pnm.cc
+++ b/caffe2/operators/lengths_reducer_ops_pnm.cc
@@ -389,10 +389,12 @@ class PNMSparseLengthsReductionOpVec : public Operator<PNMContext> {
       0; // Size of currently allocated blocked indices buffer
   uint64_t lengths_total_size_in_bytes_ =
       0; // Size of currently allocated blocked lengths buffer
+
+  // [TODO: ] move into PNMContext
   SlsDevice device_ =
-      SlsDevice::make(); // SLS device. TODO: move into PNMContext
+      SlsDevice::make(PNMDevice::Type::SLS_AXDIMM);
 
-  pnm::ContextHandler pnm_ctx_ = pnm::make_context(pnm::ContextType::SLS);
+  pnm::ContextHandler pnm_ctx_ = pnm::make_context(PNMDevice::Type::SLS_AXDIMM);
   pnm::Runner runner_{pnm_ctx_};
 
   pnm::memory::EmbeddedTablesHandler tables_ = nullptr;
diff --git a/caffe2/python/pybind_state_pnm.cc b/caffe2/python/pybind_state_pnm.cc
index 144c2b07..29263298 100644
--- a/caffe2/python/pybind_state_pnm.cc
+++ b/caffe2/python/pybind_state_pnm.cc
@@ -50,7 +50,7 @@ void addPNMGlobalMethods(py::module& m) {
          uint32_t row_size_in_bytes,
          uint32_t mem_preference,
          bool use_shared_tables) -> uintptr_t {
-        pnm_ctx = pnm::make_context(pnm::ContextType::SLS);
+        pnm_ctx = pnm::make_context(PNMDevice::Type::SLS_AXDIMM);
 
         if (use_shared_tables) {
           LOG(INFO) << "Allocated shared tables with memory preference id: "
-- 
2.34.1

